src/redux/actions/cartActions.js

```diff
@@ -0,0 +1,194 @@
+    (item) => item.id !== id
+  );
+  if (items.length === 0) {
+    dispatch(clearCart());
+  } else {
+    dispatch(removeItem(items));
+  }
+};
+export const getCartDiscount = (activeCartDiscount) => async (dispatch, getState) => {
+  const { selectedCustomer, paymentMethod } = getState().userReducer;
+  const { totalAmount, items } = getState().cartReducer;
+  const getCartDiscountPayload = setValidatePayload(
+    selectedCustomer,
+    totalAmount,
+    items,
+    paymentMethod
+  );
+  try {
+    dispatch(getDiscountRequest());
+    const promotion = await new Promise((resolve, reject) => {
+      window.Voucherify.setIdentity(selectedCustomer.source_id);
+
+      window.Voucherify.validate(getCartDiscountPayload, (response) => {
+        if (response.valid) {
+          resolve(response);
+        } else {
+          toast.error('No Cart Discount available');
+          reject(new Error(response.reason));
+        }
+      });
+    });
+    const discount = promotion.promotions.filter((promo) => promo.metadata.demostoreName === activeCartDiscount)[0];
+    dispatch(getDiscountSuccess(discount));
+    toast.success(discount.banner);
+  } catch (error) {
+    console.log('[getCartDiscount]', error);
+    dispatch(getDiscountError());
+  }
+};
+
+export const getDiscount = (voucherCode) => async (dispatch, getState) => {
+  const { selectedCustomer, paymentMethod } = getState().userReducer;
+  const { totalAmount, items } = getState().cartReducer;
+  const getDiscountPayload = setValidatePayload(
+    selectedCustomer,
+    totalAmount,
+    items,
+    paymentMethod
+  );
+  getDiscountPayload.code = voucherCode;
+  try {
+    dispatch(getDiscountRequest());
+    const discount = await new Promise((resolve, reject) => {
+      window.Voucherify.setIdentity(selectedCustomer.source_id);
+      window.Voucherify.validate(getDiscountPayload, (response) => {
+        if (response.valid) {
+          resolve(response);
+        } else {
+          if (response.error) {
+            toast.error(capitalize(response.error.message));
+          } else {
+            toast.error(capitalize(response.reason));
+          }
+          reject(new Error(response.reason));
+        }
+      });
+    });
+    dispatch(getDiscountSuccess(discount));
+    dispatch(getTotals())
+  } catch (error) {
+    console.log('[getDiscount]', error);
+    dispatch(getDiscountError());
+  }
+};
+
+export const checkoutCart = () => async (dispatch, getState) => {
+  const { selectedCustomer, paymentMethod } = getState().userReducer;
+  const { totalAmount, items, discount } = getState().cartReducer;
+  const checkoutPayload = setRedemptionPayload(
+    selectedCustomer,
+    totalAmount,
+    items,
+    paymentMethod
+  );
+  
+  // If voucher or promotion is not applied
+  if (_.isEmpty(discount)) {
+    checkoutPayload.source_id = getOrderId();
+    checkoutPayload.status = 'FULFILLED';
+
+    await sendPayload(checkoutPayload, 'order')
```

Comment:
either get rid of await and `.then()`s will handle the rest,
or get rid of `.then()`s - you're already awaiting `sendPayload()` result.

Reply:
Fixd. Thanks to this I found some other bug